src/structures/game_session.ts

```diff
@@ -947,40 +947,65 @@ export default class GameSession {
      * Creates/updates a user's activity in the data store
      * @param userID - The player's Discord user ID
      */
-    private async ensurePlayerStat(userID: string) {
-        const currentDateString = getSqlDateString();
-        await dbContext.kmq("player_stats")
-            .insert(
-                {
-                    player_id: userID,
-                    first_play: currentDateString,
-                    last_active: currentDateString,
-                },
-            )
-            .onConflict("player_id")
-            .ignore();
+    private ensurePlayerStat(userID: string) {
+        dbContext.kmq.transaction(async (trx) => {
+            const currentDateString = getSqlDateString();
+            await dbContext.kmq("player_stats")
+                .insert(
+                    {
+                        player_id: userID,
+                        first_play: currentDateString,
+                        last_active: currentDateString,
+                    },
+                )
+                .onConflict("player_id")
+                .ignore()
+                .transacting(trx);
 
-        await dbContext.kmq("player_servers")
-            .insert({
-                player_id: userID,
-                server_id: this.guildID,
-            })
-            .onConflict(["player_id", "server_id"])
-            .ignore();
+            await dbContext.kmq("temporary_player_stats")
```

Comment:
Don't need to initialize it. The `currentDateString` here and in `incrementPlayerSongsGuessed` and `incrementPlayerExp` may be different. Just insert an entry once per game session end per player

Reply:
Can I create a date object and pass it to the functions that increment song guesses, exp gain, and the initialization? It maintains good abstraction in the for loop of endSession compared to a single insert